actions/views/custom_status.ts

```diff
@@ -0,0 +1,45 @@
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {updateMe} from 'mattermost-redux/actions/users';
+import {DispatchFunc, GetStateFunc} from 'mattermost-redux/types/actions';
+
+import {CustomStatus} from 'types/store/custom_status';
+
+export function updateUserCustomStatus(newCustomStatus: CustomStatus) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const userProps = Object.assign({}, getCurrentUser(getState()).props);
+        const recentCustomStatuses = userProps.recentCustomStatuses ? JSON.parse(userProps.recentCustomStatuses) : [];
+        const updatedRecentCustomStatuses = recentCustomStatuses.filter((status: CustomStatus) => status.text !== newCustomStatus.text);
+        updatedRecentCustomStatuses.unshift(newCustomStatus);
+        if (updatedRecentCustomStatuses.length > 5) {
+            updatedRecentCustomStatuses.pop();
+        }
+        userProps.customStatus = JSON.stringify(newCustomStatus);
+        userProps.recentCustomStatuses = JSON.stringify(updatedRecentCustomStatuses);
+        const user = Object.assign({}, getCurrentUser(getState()), {props: userProps});
```

Comment:
Again, why `Object.assign`?

Also, you can use `getCurrentUser(getState())` just once.